src/core/toga/window.py

```diff
@@ -270,27 +270,34 @@ def open_file_dialog(self, title, initial_directory=None, file_types=None, multi
         """ This opens a native dialog where the user can select the file to open.
         It is possible to set the initial folder and only show files with specified file extensions.
         If no path is returned (eg. dialog is canceled), a ValueError is raised.
+
         Args:
-            title (str): The title of the dialog window.
-            initial_directory(str): Initial folder displayed in the dialog.
-            file_types: A list of strings with the allowed file extensions.
+            title (str): The title of the dialog window (ignored on Android)
+            initial_directory(str): Initial folder displayed in the dialog. On Android, this needs to be a content URI,
+                e.g. 'content://com.android.externalstorage.documents/document/primary%3ADownload%2FTest-dir'
+            file_types: A list of strings with the allowed file extensions. On Android, these must be MIME types,
+                e.g. ['application/pdf','application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'].
             multiselect: Value showing whether a user can select multiple files.
 
         Returns:
-            The absolute path(str) to the selected file or a list(str) if multiselect
+            The absolute path(str) to the selected file or a list(str) if multiselect. On Android, you will get back
```

Comment:
I understand that URIs are required for Android support, but specific references to Android in the platform-agnostic API layer isn't desirable. 

We should either (a) double down on this, say the return value here is "a file path or URI", and expect the user to support both; or (b) say that it returns a path, and leave it as a footnote that this will be a URI in some contexts. (a) is committing to a permanent solution; (b) would a precursor to a longer term "file access API" that we discussed in chat.

I *think* I'm leaning towards (b), even if we can only get as far as documenting the longer term plan as a ticket.